### PR TITLE
Allow for horizontal layout option

### LINKED ISSUE #1060: Allow for new Example presentation on certain pages
On certain pages, we may want to allow for a different Example presentation as can be seen in these designs: https://www.figma.com/file/IvYbGVlucoUMvUgbYFSG7a/Design-System-Card-Site?node-id=813%3A239

### PR DESCRIPTION
**Pull request by taysea**
[Branch Preview](https://deploy-preview-1076--keen-mayer-a86c8b.netlify.app/components/card)

#### What does this PR do?
Designs: https://www.figma.com/file/IvYbGVlucoUMvUgbYFSG7a/Design-System-Card-Site?node-id=813%3A239 (should be in WIP page)

Introduces `horizontalLayout` and `plain` as props that can be added to the `Example` component in order to meet requirements of new example designs. When `horizontalLayout` is applied, the visual and the code/resources will appear in a row instead of column. If `plain` is applied, the outer container/frame of the example will be stripped -- just leaving the example content. This may be desired on certain component examples such as Card.

#### Where should the reviewer start?
aries-site/src/layouts/content/Example/Example.js

#### How should this be manually tested?
On Card page deploy. Both `horizontalLayout` and `plain` have been applied.

#### What are the relevant issues?
Closes #1060 

#### Should this PR be mentioned in Design System updates?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes, backwards compatible.

### CODE REVIEW

File: aries-site/src/layouts/content/Example/Container.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  plain,
+  screenContainer,
+  showResponsiveControls,
+  template,
+  ...rest
+}) => {
+  // Height for template screen needs to be between medium and large
+  // to maintain aspect ratio, so this is small + medium
+  const { small, medium } = defaultProps.theme.global.size;
+  const aspectHeight = `${parseInt(medium, 10) + parseInt(small, 10)}px`;
```
Comment (halocline) [inline aries-site/src/layouts/content/Example/Container.js:21]: I wonder if this is more straight forward?
```suggestion
  const aspectHeight = `${Number(medium) + Number(small)}px`;
```

File: aries-site/src/layouts/content/Example/Container.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Box, defaultProps } from 'grommet';
+
+// This creates the overall canvas/frame that the Example sits inside of
+export const Container = ({
```
Comment (halocline) [inline aries-site/src/layouts/content/Example/Container.js:6]: This is a nice change. I like having the `<Container />`

File: aries-site/src/layouts/content/Example/Container.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  designer,
+  docs,
+  figma,
+  height: heightProp,
+  horizontalLayout,
+  plain,
+  screenContainer,
+  showResponsiveControls,
+  template,
```
Comment (halocline) [inline aries-site/src/layouts/content/Example/Container.js:15]: With the absence of docs, I wonder if we could add comments here (or wherever makes sense) which briefly explains what each of these props are/do? Thoughts?

Comment (halocline) [reply aries-site/src/layouts/content/Example/Container.js:15]: Maybe this makes more sense on Example.js?

File: aries-site/src/layouts/content/Example/Container.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  ...rest
+}) => {
+  // Height for template screen needs to be between medium and large
+  // to maintain aspect ratio, so this is small + medium
+  const { small, medium } = defaultProps.theme.global.size;
+  const aspectHeight = `${parseInt(medium, 10) + parseInt(small, 10)}px`;
+
+  let height;
+  if (template || screenContainer) height = aspectHeight;
+  else if (!plain) height = { min: 'medium' };
```
Comment (halocline) [inline aries-site/src/layouts/content/Example/Container.js:25]: how does this play with the defaultProps below? Are one of these redundant or irrelevant?

File: aries-site/src/layouts/content/Example/ExampleControls.js [hunk 5]
```diff
@@ -3,11 +3,76 @@ import PropTypes from 'prop-types';
...
+                title="Open docs"
+                a11yTitle="Open docs"
+                href={docs}
+                icon={<Document />}
+                target="_blank"
+                size={buttonSize}
+              />
+            </Box>
+          )}
+          <Button
```
Comment (halocline) [inline aries-site/src/layouts/content/Example/ExampleControls.js:64]: This seems to be the same as lines 91-125-ish. Any opportunities to simplify/reuse this?

File: aries-site/src/examples/components/card/CardAnatomy.js [hunk 6]
```diff
@@ -6,7 +6,7 @@ import { Info, Emoji } from 'grommet-icons';
 const spacing = defaultProps.theme.global.edgeSize;
 
 export const CardAnatomy = () => (
-  <Card width="medium">
+  <Card background="background" width="medium">
```
Comment (halocline) [inline aries-site/src/examples/components/card/CardAnatomy.js:9]: I missed this in last week's Card PR. I think we have opportunities to set some default backgrounds and padding in the theme. I have those defined and will submit in a separate PR along with updates to these examples for cleanup purposes.

For the scope of this PR, can we change this to:
```suggestion
  <Card background="background-front" width="medium">
```

This will help with visibility on Dark mode.

File: aries-site/src/examples/components/card/CardExample.js [hunk 7]
```diff
@@ -34,6 +34,7 @@ export const CardExample = () => {
 
   return (
     <StyledCard
+      background="background"
```
Comment (halocline) [inline aries-site/src/examples/components/card/CardExample.js:37]:
```suggestion
      background="background-front"
```

File: aries-site/src/examples/components/card/CardMeterExample.js [hunk 8]
```diff
@@ -7,6 +7,7 @@ const mockAvailability = 0.872;
 
 export const CardMeterExample = () => (
   <Card
+    background="background"
```
Comment (halocline) [inline aries-site/src/examples/components/card/CardMeterExample.js:10]:
```suggestion
    background="background-front"
```

File: aries-site/src/examples/components/card/CardWithFooter.js [hunk 9]
```diff
@@ -4,7 +4,7 @@ import { Box, Card, CardFooter, Text } from 'grommet';
 import { Alert, Server } from 'grommet-icons';
 
 export const CardWithFooter = () => (
-  <Card width="medium">
+  <Card background="background" width="medium">
```
Comment (halocline) [inline aries-site/src/examples/components/card/CardWithFooter.js:7]:
```suggestion
  <Card background="background-front" width="medium">
```

File: aries-site/src/examples/components/card/CardWithImage.js [hunk 10]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { Box, Card, CardBody, Image, Text } from 'grommet';
 
 export const CardWithImage = () => (
-  <Card>
+  <Card background="background">
```
Comment (halocline) [inline aries-site/src/examples/components/card/CardWithImage.js:5]:
```suggestion
  <Card background="background-front">
```

File: aries-site/src/examples/components/card/SimpleCard.js [hunk 11]
```diff
@@ -3,7 +3,7 @@ import { Box, Card, Text } from 'grommet';
 import { Hpe } from 'grommet-icons';
 
 export const SimpleCard = () => (
-  <Card pad="medium" width="medium">
+  <Card background="background" pad="medium" width="medium">
```
Comment (halocline) [inline aries-site/src/examples/components/card/SimpleCard.js:6]:
```suggestion
  <Card background="background-front" pad="medium" width="medium">
```

File: aries-site/src/layouts/content/Example/Container.js (hunk 4 above)
Comment (taysea) [reply aries-site/src/layouts/content/Example/Container.js:25]: Good catch, the default props was redundant.

File: aries-site/src/layouts/content/Example/ExampleControls.js (hunk 5 above)
Comment (taysea) [reply aries-site/src/layouts/content/Example/ExampleControls.js:64]: Yes, definitely. Implementing on upcoming commit.

File: aries-site/src/layouts/content/Example/Container.js (hunk 3 above)
Comment (taysea) [reply aries-site/src/layouts/content/Example/Container.js:15]: I'll add the comments on `Example.js` then reference that here.
